src/BloomExe/Edit/JpegWarningDialog.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Windows.Forms;
+using L10NSharp;
+
+namespace Bloom.Edit
+{
+	public partial class JpegWarningDialog : Form
+	{
+		public JpegWarningDialog()
+		{
+			InitializeComponent();
+			_warningText.Text = LocalizationManager.GetString("EditTab.JpegWarningDialog.WarningText",
+				@"The file you’ve chosen is a “jpeg” file. JPEGs are perfect for photographs and color artwork. However, JPEGs are a big problem for black and white line art. Problems include:
+• Fuzziness and grey dots.
+• Large file sizes, making the book hard to share.
+• If there are many large JPEGs, Bloom may not have enough memory to make PDFs.
+
+Note: Because JPEG is “lossy”, converting a JPEG to PNG, TIFF, or bmp actually makes things even worse. If this is black and white line art, you want to get an original scan in one of those formats.
+
+Please select from one of the following, then click “OK”:").Replace("\r\n","\n").Replace("\n", Environment.NewLine);
+		}
+
+		private void _okButton_Click(object sender, EventArgs e)
+		{
+			DialogResult = _cancelRadioButton.Checked ? DialogResult.Cancel : DialogResult.OK;
+			Close();
+		}
+
+		/// <summary>
+		/// Tells whether the supplied image looks suspiciously like a file that 
+		/// would be better off as a PNG. Not particularly smart, but fast.
+		/// </summary>
+		public static Boolean ShouldWarnAboutJpeg(Image image)
+		{
+			var bmp = image as Bitmap;
+			if (bmp == null)
+				return false;
+			return HasLotsOfWhiteSpace(bmp) && (IsGreyScale(bmp) || !HasLotsOfColor(bmp));
+		}
+		private static bool IsGreyScale(Bitmap bmp)
+		{
+			var sampleLinePercentages = new[] { 20, 50, 70 };
+			return
+				sampleLinePercentages.Any(sampleLine => GetIsGrey(bmp, (int)(bmp.Height * (sampleLine / 100.0))));
+		}
+		private static bool HasLotsOfColor(Bitmap bmp)
```

Comment:
This cannot succeed for any image less than 100 pixels wide. Should we take line length into consideration at all?